compiler/backend/src/org/jetbrains/kotlin/codegen/inline/InlineCodegen.kt

```diff
@@ -206,9 +207,9 @@ abstract class InlineCodegen<out T : BaseExpressionCodegen>(
                 ?: error("No stack value for continuation parameter of suspend function")
     }
 
-    protected fun inlineCall(nodeAndSmap: SMAPAndMethodNode, inlineDefaultLambda: Boolean): InlineResult {
+    protected fun inlineCall(nodeAndSmap: SMAPAndMethodNode, inlineDefaultLambda: Boolean, isSyntheticCall: Boolean): InlineResult {
         assert(delayedHiddenWriting == null) { "'putHiddenParamsIntoLocals' should be called after 'processAndPutHiddenParameters(true)'" }
-        defaultSourceMapper.callSiteMarker = CallSiteMarker(codegen.lastLineNumber)
+        if (!isSyntheticCall) defaultSourceMapper.callSiteMarker = CallSiteMarker(codegen.lastLineNumber)
```

Comment:
If I undestand correctly it should work in case of inlining original function to $default companion one? If so maybe check container origin in this case instead of `isSyntheticCall = expression.startOffset == UNDEFINED_OFFSET`?

Reply:
I believe this is when inlining the $default companion into the call site of the original. That is, in the source is a call `foo`, it gets redirected to a call to `foo$default` and this is then inlined.

Reply:
So on inlining `foo$default` we shouldn't reset markers, that looks strange. I thought that we do smth wrong on inliniing foo into foo$default

Reply:
Even in such case i think  it's possible  to check origin for FUNCTION_FOR_DEFAULT_PARAMETER. 
But I still confused in difference between general and $default function processing in ir-inliner